QuizApp/QuizApp/DataSources/QuizDatabaseDataSource.swift

```diff
@@ -0,0 +1,37 @@
+import RealmSwift
+
+protocol QuizDatabaseDataSourceProtocol {
+
+    func fetchQuizzes() -> [QuizDatabaseModel]
+
+    func save(quizzes: [QuizDatabaseModel])
+
+}
+
+class QuizDatabaseDataSource: QuizDatabaseDataSourceProtocol {
+
+    func fetchQuizzes() -> [QuizDatabaseModel] {
+        guard let realm = try? Realm() else { return [] }
```

Comment:
Create a class var `private let realm: Realm?` and assign a value to it on `init`. You can guard it here.